images/metering-ansible-operator/roles/meteringconfig/tasks/configure_storage.yml

```diff
@@ -36,6 +36,36 @@
 
   when: meteringconfig_storage_hive_storage_type == 's3' and meteringconfig_storage_s3_create_bucket
 
+- name: Configure S3 Compatible Storage
+  block:
+    - name: Validate Metering s3Compatible bucket name
+      assert:
+        that:
+          - meteringconfig_storage_s3Compatible_bucket_name != ""
+        msg: "storage.hive.s3Compatible.bucket cannot be empty"
+
+    - name: Validating nonempty endpoint
+      assert:
+        that:
+          - meteringconfig_storage_s3Compatible_endpoint != ""
+        msg: "storage.hive.s3Compatible.endpoint cannot be empty"
+
+    - name: Validate Metering s3Compatible credentials
+      assert:
+        that:
+          - meteringconfig_storage_s3Compatible_credentials_secret_name == ""
+        msg: "storage.hive.s3Compatible.secretName is only used to reference existing secrets, must be empty if creating new secrets"
+      when: meteringconfig_storage_s3Compatible_create_secret
+
+    - name: Validate Metering s3Compatible credentials
```

Comment:
update this task name to reflect that you're validating that `secretName: ""` when `createSecret: false`